File: src/main/java/domain/RookImpl.java
package domain;

import domain.model.Cell;
import domain.model.CellKt;

/**
 * Реализация {@link Rook}.
 */
public class RookImpl extends Rook {
    /**
     * Текущая позиция.
     */
    private Cell position;

    @Override
    public void onStart(Cell position) {
        this.position = position;
    }

    @Override
    public Cell makeMove(Cell kingPosition) {
        //TODO реализовать логику рандомной ходьбы по доске
        position = CellKt.right(position);
        return position;
    }
}

File: src/main/java/domain/RookImpl.kt
package domain

import domain.model.*
import kotlin.random.Random

/**
 * Реализация [Rook].
 */
class RookImpl : Rook() {

    /**
     * Текущая позиция.
     */
    private lateinit var rookPosition: Cell

    override fun onStart(position: Cell) {
        rookPosition = position
    }

    override fun makeMove(kingPosition: Cell): Cell {
        val gameBoardRange = (0 until GameBoard.size)
        var newRookPosition = getRandomMove()
        while (!newRookPosition.isInRange((gameBoardRange)) || newRookPosition == kingPosition) {
            newRookPosition = getRandomMove()
        }
        rookPosition = newRookPosition
        return rookPosition
    }

    private fun getRandomMove(): Cell = when (Random.nextInt(4)) {
        0 -> rookPosition.right
        1 -> rookPosition.left
        2 -> rookPosition.up
        else -> rookPosition.down
    }
}